src/Materials/Textures/Packer/packerLoader.ts

```diff
@@ -0,0 +1,71 @@
+import { TexturePacker, ITexturePackerOptions } from "./packer";
+import { Scene } from "../../../scene";
+
+/**
+ * Defines the basic interface of a TexturePacker JSON File
+ */
+export interface ITexturePackerJSON{
+
+    /**
+	 * The frame ID
+	 */
+    name: string;
+
+    /**
+	 * The base64 channel data
+	 */
+    sets: any;
+
+    /**
+	 * The options of the Packer
+	 */
+    options: ITexturePackerOptions;
+
+    /**
+	 * The frame data of the Packer
+	 */
+    frames: Array<number>;
+
+}
+
+/**
+ * This is a support class to load JSON texture packer data.
+ * @see #TODO ADD THIS
+ */
+export class TexturePackerLoader{
+    /**
+    * Initializes a XML request for the JSON file.
+    * @param jsonURL The URL of the texturePackage.json
+    * @param scene that the packer is scoped to.
+    * @returns TexturePacker
+    */
+    constructor(jsonURL: string, scene: Scene) {
+        if (!scene || !jsonURL || typeof jsonURL !== 'string') {
+            return false;
+        }
+        try {
+            let pack = new TexturePacker('Parsing_TP', [], {}, scene);
+            pack.promise = new Promise ((success, error) => {
+            try {
+                var xml = new XMLHttpRequest();
+                xml.onreadystatechange = () => {
+                    if (xml.readyState == 4) {
+                        if (xml.status == 200) {
+                            pack.updateFromJSON(xml.responseText, success, error);
+                        }
+                    }
+                };
+                xml.open("GET", jsonURL);
+                xml.send();
+                }catch (e) {
+                    error(e);
+                }
+            });
+
+            return pack;

```

Comment:
Not sure to understand how the user is supposed to get the pack variable.

Overall I feel like we do not want this class.

Reply:
I was accessing it in the "Then" function to make sure it was compiled.

Reply:
What would be an alternative off the top of your head?

If we change it from a promise then to a async Promise then I will have a onComplete method correct?

Reply:
The alternative it to use the loadFromJSON directly and let the user deal with where the data comes from

Reply:
```
pack = new BABYLON.TexturePackerLoader('./textures/TestPack_texurePackage.json', scene)    
    pack.then(()=>{
        console.log("Pack Parsed")
        console.log("texturePack", pack)        
        let packMat = new BABYLON.StandardMaterial('packMat', scene )
        output.material = packMat;
        pack.setMeshToFrame(output, 2, true)
    }, (error)=>{
        console.log(error)
    }) 
```

Was working on accessing the pack variable, but maybe I am doing this incorrect though from what you are saying.  Let me review everything you have said and make sure Im not being dumb.